### ISSUE TITLE
Training won't start -- torch distributed error -- even with one GPU selected

### ISSUE THREAD

**Issue by gessyoo** [SayanoAI/RVC-Studio#25]
I was able to train one model successfully, but for reasons I can't understand, all subsequent attempts fail to train, even after a re-install from scratch.  I'm attaching a screencap of the error message.  This github thread https://github.com/Lightning-AI/pytorch-lightning/discussions/11865 suggests that the error could be solved  by setting an environment variable -- `
`SET` export `NCCL_P2P_DISABLE=1` but that variable doesn't seem to persist after running streamlit run Home.py.

Is there a solution that I can implement in the code?

![torch_distributed_error](https://github.com/SayanoAI/RVC-Studio/assets/35972345/0726e60d-d6d5-4f56-9b7b-73b5278a518f)

**Comment by SayanoAI** [comment]
Try the latest commit. I've added os.environ["NCCL_P2P_DISABLE"] = 1 to line 12 of the training_cli.py file so hopefully that fixes it.

**Comment by gessyoo** [comment]
I tried the latest commit, still training wouldn't start.  But I really like that your project periodically saves the "best model," during the training, so I kept digging around for a solution.  The solution I found involved slightly modifying lines 10 through 14 of training_cli.py as follows:
``
hps = utils.get_hparams()
os.environ["CUDA_VISIBLE_DEVICES"] = hps.gpus.replace("-", ",")
n_gpus = len(hps.gpus.split("-"))
from random import shuffle, randint
import traceback, json, argparse, itertools, math, torch, pdb
``
with this modification, training starts as it should and runs normally on my RTX a6000.  Posting in case it might help someone else with the same issue.
``
